pkg/v1/tarball/image.go

```diff
@@ -216,6 +218,10 @@ func extractFileFromTar(opener Opener, filePath string) (io.ReadCloser, error) {
 			return nil, err
 		}
 		if hdr.Name == filePath {
+			if hdr.Typeflag == tar.TypeSymlink || hdr.Typeflag == tar.TypeLink {
+				currentDir := filepath.Dir(filePath)
```

Comment:
I have one concern about using `filepath.Dir` here. I'm not familiar with e.g. windows, but would this work when running on a windows machine if the tarball has normal path separators?

I think it's fine based on [this](https://golang.org/src/os/path_windows.go#L12) but I'd love some confirmation.

Reply:
I just tested in Windows, works same.